### ISSUE TITLE
Unexpectedly found nil while unwrapping an Optional value

### ISSUE THREAD

**Issue by kevvdevv** [stleamist/BetterSafariView#19]
I am getting the "Unexpectedly found nil while unwrapping an Optional value" error when trying to call webAuthenticationSession in a SwiftUI project.  The error points to line 185 in WebAuthenticationPresenter.swift.

```
func presentationAnchor(for session: ASWebAuthenticationSession) -> ASPresentationAnchor {
                return coordinator!.viewController.view.window!
            }
```

My code looks like this:

```
struct OnBoardingView: View {
    @State var showAuth = true
    var body: some View {
        checkInoReaderAuth()
            .webAuthenticationSession(isPresented: $showAuth) {
                WebAuthenticationSession(url: [my url], callbackURLScheme: nil) { url, error in
                    webAuthSessionClosure(url, error)
                }
            }
    }
}
```

Welcome any suggestions.

**Comment by jonthornham** [comment]
I am getting the same error. I am generating a URL and setting it in a state variable. Once the string is set I tap a button to load the BetterSafariView and I get "Unexpectedly found nil while unwrapping an Optional value" The code below shows two button. One to get the link to verify the link is there. The second to load the view. 

```
Button(action: {
                            self.createStripeConnectAccountLink() {result in
                                if result == true {
                                    print(self.stripeAccountLink)
                                }
                            }
                                }) {
                                    Text("Get Link")
                                }
                        Button(action: {
                            self.createStripeConnectAccountLink() {result in
                                self.isWebViewPresented.toggle()
                            }
                                }) {
                                    Text("Present SafariView")
                                }
                        .safariView(isPresented: $isWebViewPresented) {
                            SafariView(
                                url: URL(string: stripeAccountLink)!,
                                configuration: SafariView.Configuration(
                                    entersReaderIfAvailable: false,
                                    barCollapsingEnabled: true
                                )
                            )
                            .preferredBarAccentColor(.pacificBlueAccent)
                            .preferredControlAccentColor(.white)
                            .dismissButtonStyle(.done)
                        }
```

**Comment by jonthornham** [comment]
It appears the issue occurs anytime a direct link is not added. For example this works. 

```
Button(action: {
                            self.createStripeConnectAccountLink() {result in
                                self.isWebViewPresented.toggle()
                            }
                                }) {
                                    Text("Present SafariView")
                                }
                        .safariView(isPresented: $isWebViewPresented) {
                            SafariView(
                                url: URL(string: "http:www.github.com"!,
                                configuration: SafariView.Configuration(
                                    entersReaderIfAvailable: false,
                                    barCollapsingEnabled: true
                                )
                            )
                            .preferredBarAccentColor(.pacificBlueAccent)
                            .preferredControlAccentColor(.white)
                            .dismissButtonStyle(.done)
                        }
```

This does not.

```
Button(action: {
                            self.createStripeConnectAccountLink() {result in
                                self.isWebViewPresented.toggle()
                            }
                                }) {
                                    Text("Present SafariView")
                                }
                        .safariView(isPresented: $isWebViewPresented) {
                            SafariView(
                                url: URL(string: stripeAccountLink)!,
                                configuration: SafariView.Configuration(
                                    entersReaderIfAvailable: false,
                                    barCollapsingEnabled: true
                                )
                            )
                            .preferredBarAccentColor(.pacificBlueAccent)
                            .preferredControlAccentColor(.white)
                            .dismissButtonStyle(.done)
                        }
```

**Comment by kevvdevv** [comment]
In digging into this a bit further, I'd propose changing 

`coordinator!.viewController.view.window!`

to 

`ASPresentationAnchor()`

**Comment by stleamist** [comment]
This issue has been fixed in [v2.4.1](https://github.com/stleamist/BetterSafariView/releases/tag/v2.4.1), as a result of #22.

Thank you for your contribution and waiting :)
